custom/abt/reports/late_pmt.py

```diff
@@ -139,66 +80,85 @@ def headers(self):
         )
 
     @cached_property
-    def smss_received(self):
-        data = SMS.objects.filter(
-            domain=self.domain,
-            couch_recipient_doc_type='CommCareUser',
-            direction=INCOMING,
-            couch_recipient__in=self.get_user_ids,
-            date__range=(
-                self.startdate,
-                self.enddate
-            )
-        ).exclude(
-            text="123"
-        ).values('date', 'couch_recipient').annotate(
-            number_of_sms=Count('couch_recipient')
-        )
-        return {(sms['date'].date(), sms['couch_recipient']) for sms in data}
-
-    @cached_property
-    def get_user_ids(self):
-        return [user['user_id'] for user in self.get_users]
-
-    @cached_property
-    def get_users(self):
-        return LatePMTUsers(config=self.report_config).get_data()
+    def pmts_submitted(self):
+        forms = list(iter_forms_by_xmlns_received_on(
+            self.domain, INDICATORS_FORM_XMLNS, self.startdate, self.enddate
+        ))
+        return {
+            (case.received_on, case.operation_site)
+            for case in get_cases_from_forms(self.domain, forms)
+        }
 
     @property
     def rows(self):
-        def _to_report_format(date, user, error_msg):
+        def _to_report_format(date, location, error_msg):
             return [
                 date.strftime("%Y-%m-%d"),
-                user['username'].split('@')[0],
-                user['phone_number'],
-                user['country'],
-                user['level_1'],
-                user['level_2'],
-                user['level_3'],
-                user['level_4'],
+                location.name,
+                location.country,
+                location.level_1,
+                location.level_2,
+                location.level_3,
+                location.level_4,
                 error_msg
             ]
 
-        users = self.get_users
+        include_missing_pmt_data = self.report_config['submission_status'] != 'group_b'
+        # include_incorrect_pmt_data is no longer applicable, because an
+        # invalid SMS will not result in a pmt_data case being created/updated
```

Comment:
Can the filter just be removed from the report?

Reply:
:thinking: That would be nice. I've removed the unused option now. I'll check whether we can get rid of the filter completely.